### PR TITLE
WIP: Improve shellout logging and add timeout

### PR DESCRIPTION
**Pull request by tas50**
Log successful and failed commands. This gives you the necessary context to dig into issues

### CODE REVIEW

File: lib/ohai/mixin/command.rb [hunk 1]
```diff
@@ -28,10 +28,29 @@
...
   module Mixin
     module Command
-      def shell_out(cmd)
-        m = Mixlib::ShellOut.new(cmd)
-        m.run_command
-        m
+      # accept a command and any of the mixlib-shellout options
+      def shell_out(cmd, **options)
+        # unless specified by the caller timeout after 30 seconds
+        options[:timeout] = 30 unless options[:timeout]
```
Comment (lamont-granquist) [inline lib/ohai/mixin/command.rb]: `options[:timeout] ||= 30`

**Comment by lamont-granquist** [comment]
tiny niggle, LGTM

File: lib/ohai/mixin/command.rb (hunk 1 above)
Comment (tas50) [inline lib/ohai/mixin/command.rb]: That would make a lot more sense

File: lib/ohai/mixin/command.rb [hunk 2]
```diff
@@ -28,10 +28,29 @@
...
+        m = Mixlib::ShellOut.new(cmd, options)
+        begin
+          m.run_command
+
+          # we should really fail here on non-0, but historically we haven't so we can't now
```
Comment (mcquin) [inline lib/ohai/mixin/command.rb]: I believe there is one plugin that collects on non-zero. :)

**Comment by mcquin** [comment]
Generally, I like to caution against using one or two letter variable names in Ruby. You can get yourself into weird shadowing bugs.

File: lib/ohai/mixin/command.rb (hunk 2 above)
Comment (tas50) [inline lib/ohai/mixin/command.rb]: Good point.  I'll fix that one.

**Comment by tas50** [comment]
Good point. I'll fix that one.

**Comment by mcquin** [comment]
Closing in favor of https://github.com/chef/ohai/pull/788

**Pull request closed without merging**